### ISSUE TITLE
Conditional formatting

### ISSUE THREAD

**Issue by Sweenu** [Alexays/Waybar#689]
I'd like to have the ability to change the format section depending on certain criterias.
For example, in the `sway/workspaces` module, I want to have:
- `format: "{name}"` in general
- `format: "{icon}"` if the workspace's name is `slack` or `spotify`

Is something like that already possible?

**Comment by arran-nz** [comment]
Oh do I have the project for you: https://github.com/Lyr-7D1h/swayest_workstyle

**Comment by tv42** [comment]
Another use case: Niri workspaces an index and sometimes a name. The built-in `{value}` shows name if set, index if not. It is impossible for me to configure output that shows the index *and* the name if set, without having trailing junk when the name is not set (`{index}:{name}` shows `6:`; even trailing whitespace is noticeable, but a colon makes it more obvious).

https://github.com/Alexays/Waybar/wiki/Module:-Niri#format-replacements

**Comment by Alexays** [comment]
Yes, this is possible for `sway/workspaces` using `format-icons` keyed by workspace name plus `format: "{icon}"`. Icons are selected by the workspace name, and anything not matched falls back to `default`:

```jsonc
"sway/workspaces": {
    "format": "{icon}",
    "format-icons": {
        "slack": "",
        "spotify": "",
        "default": "{name}"
    }
}
```

So named workspaces show their icon and everything else shows `{name}` (see `Workspaces::getIcon` in `src/modules/sway/workspaces.cpp`). Closing, but reopen if you need something more advanced. 🙏

**Comment by baodrate** [comment]
> and everything else shows `{name}` (see `Workspaces::getIcon` in `src/modules/sway/workspaces.cpp`).

Is this expected to work in the current stable release? I have the same setup[^1] and waybar shows the literal `{name}` rather than the name of the workspace.

[^1]: my actual config:

    ```
    "sway/workspaces" = {
      all-outputs = false;
      disable-scroll = true;
      format = "{icon}";
      format-icons = {
        "1" = "一";
        "2" = "二";
        "3" = "三";
        "4" = "四";
        "5" = "五";
        "6" = "六";
        "7" = "七";
        "8" = "八";
        "9" = "九";
        "10" = "十";
        default = "{name}";
      };
    };
    ```
